.travis.yml

```diff
@@ -20,166 +17,144 @@ sudo: false
         # the Numpy developers have taken care of testing Numpy with different
         # versions of Python, we can vary Python and Numpy versions at the same
         # time.
-        # Run this test using native pytest
         - os: linux
-          env: PYTHON_VERSION=3.5 NUMPY_VERSION=1.13
-               INSTALL_CMD='python setup.py build_ext --inplace'
-               PIP_DEPENDENCIES='pytest-astropy'
-               TEST_CMD='pytest --open-files --doctest-rst'
-               PYTEST_VERSION=3.7
-          script:
-            - $INSTALL_CMD
-            - $TEST_CMD
+          python: 3.5
+          env: TOXENV=py35
+               TOXOPTS='--develop --force-dep \'pytest==3.7\''
+               TOXARGS='--cov=astropy --open-files --doctest-rst'
+               CFLAGS='--coverage -fno-inline-functions -O0'
 
         # Now try with all optional dependencies. We also include the --readonly
         # flag to make sure no files are being written to the temporary install
         # location during testing.
         - os: linux
-          env: PYTHON_VERSION=3.5 SETUP_CMD='test --remote-data=astropy --readonly'
-               CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
-               PIP_DEPENDENCIES="jplephem asdf"
-               LC_CTYPE=C.ascii LC_ALL=C
-               PYTEST_VERSION=3.7
+          python: 3.5
+          env: TOXENV=py35-alldeps
+               TOXOPTS='--force-dep \'pytest==3.7\''
+               TOXARGS='--cov=astropy --remote-data=astropy'
+               CFLAGS='--coverage -fno-inline-functions -O0'
+               # TOXARGS='--remote-data=astropy --readonly'
+               # LC_CTYPE=C.ascii LC_ALL=C
+               # PIP_DEPENDENCIES="asdf"
 
         - os: linux
           stage: Initial tests
-          env: PYTHON_VERSION=3.7 CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
-               PIP_DEPENDENCIES="asdf"
-               SETUP_CMD='test -a "--durations=50"'
+          python: 3.7
+          dist: xenial
+          sudo: true
+          env: TOXENV=py37-alldeps
+               TOXARGS='--durations=50'
+               # PIP_DEPENDENCIES="asdf"
           compiler: clang
 
         # Full tests with coverage checks.
         - os: linux
-          env: SETUP_CMD='test --coverage --remote-data=astropy --readonly'
-               CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
-               PIP_DEPENDENCIES="codecov objgraph jplephem bintrees sortedcontainers asdf"
-               LC_CTYPE=C.ascii LC_ALL=C
+          python: 3.6
+          env: TOXENV=py36-alldeps
+               TOXARGS='--cov=astropy --remote-data=astropy'
                CFLAGS='--coverage -fno-inline-functions -O0'
-               MATPLOTLIB_VERSION=2.0
                EVENT_TYPE='push pull_request cron'
+               # TOXARGS='--remote-data=astropy --readonly'
+               # LC_CTYPE=C.ascii LC_ALL=C
+               # PIP_DEPENDENCIES="objgraph bintrees sortedcontainers asdf"
+               # MATPLOTLIB_VERSION=2.0
```

Comment:
Fine to remove the MPL pinning - we have version specific testing of matplotlib on CircleCI